service/src/main/java/bio/terra/axonserver/service/file/FileService.java

```diff
@@ -3,91 +3,100 @@
+ * file to a desired format.
+ */
 @Component
 public class FileService {
 
-  WorkspaceManagerService wsmService;
-  ConvertService convertService;
+  private final WorkspaceManagerService wsmService;
+  private final ConvertService convertService;
+
+  private record FileWithName(byte[] file, String fileName) {}
 
   @Autowired
   public FileService(WorkspaceManagerService wsmService, ConvertService convertService) {
     this.wsmService = wsmService;
     this.convertService = convertService;
   }
 
+  /**
+   * Gets a file from a given resource. Optionally converts the file to a desired format.
+   *
+   * @param user The SAM user requesting the file
+   * @param workspaceId The workspace that the resource is in
+   * @param resourceId The id of the resource that the object is in
+   * @param objectPath The path to the object in the bucket. Only used if the resource is a bucket.
+   * @param convertTo The format to convert the file to. If null, the file is not converted.
+   * @return The file as a byte array
+   */
   public ByteArrayResource getFile(
-      AuthenticatedUserRequest userRequest,
+      SamUser user,
       UUID workspaceId,
       UUID resourceId,
       @Nullable String objectPath,
       @Nullable String convertTo) {
 
-    ResourceDescription resource = this.getResource(userRequest, workspaceId, resourceId);
-
-    Pair<byte[], String> fileWithName = this.getFileHandler(resource, objectPath, userRequest);
-    byte[] file = fileWithName.getValue0();
-    String fileName = fileWithName.getValue1();
+    ResourceDescription resource = this.getResource(user, workspaceId, resourceId);
 
+    FileWithName fileWithName = this.getFileHandler(resource, objectPath, user);
+    byte[] file = fileWithName.file;
     if (convertTo != null) {
-      String fileExtension = FilenameUtils.getExtension(fileName);
-      file = this.convertService.convert(file, fileExtension, convertTo, userRequest);
+      String fileExtension = FilenameUtils.getExtension(fileWithName.fileName);
+      file = this.convertService.convertFile(file, fileExtension, convertTo, user);
     }
     return new ByteArrayResource(file);
   }
 
-  private Pair<byte[], String> getFileHandler(
-      ResourceDescription resource,
-      @Nullable String objectPath,
-      AuthenticatedUserRequest userRequest) {
+  private FileWithName getFileHandler(
+      ResourceDescription resource, @Nullable String objectPath, SamUser userRequest) {
 
     return switch (resource.getMetadata().getResourceType()) {
-      case GCS_OBJECT -> {
-        String objectName = resource.getResourceAttributes().getGcpGcsObject().getFileName();
-        yield new Pair<byte[], String>(this.getGcsObjectFile(resource, userRequest), objectName);
-      }
-      case GCS_BUCKET -> new Pair<byte[], String>(
-          this.getGcsBucketFile(resource, objectPath, userRequest), objectPath);
+      case GCS_OBJECT -> this.getGcsObjectFile(resource, userRequest);
```

Comment:
same here, if else is enough.

Reply:
There's 3 cases here, and likely more to come. I think I'll leave this one as is if thats okay.